src/tls/settings.rs

```diff
@@ -187,6 +142,72 @@ impl TlsSettings {
     }
 }
 
+impl TlsOptions {
+    fn load_authorities(&self) -> Result<Vec<X509>> {
+        match &self.ca_path {
+            None => Ok(vec![]),
+            Some(filename) => {
+                let data = open_read(filename, "certificate")?;
+                der_or_pem(
+                    data,
+                    |der| X509::from_der(&der).map(|x509| vec![x509]),
+                    |pem| {
+                        pem.match_indices(PEM_START_MARKER)
+                            .map(|(start, _)| X509::from_pem(pem[start..].as_bytes()))
```

Comment:
I assume from this that the parser knows when to stop on its own and we don't need to limit its view of the input data?

Reply:
As far as I am aware, yes, and evidence says yes too, otherwise parsing files containing multiple certificates would have failed completely. Instead, it used to load just the first certificate.